tf2onnx/utils.py

```diff
@@ -121,17 +121,20 @@ def tf_to_onnx_tensor(tensor, name=""):
     new_type = TF_TO_ONNX_DTYPE[tensor.dtype]
     tdim = tensor.tensor_shape.dim
     dims = [d.size for d in tdim]
-    # FIXME: something is fishy here
-    if dims == [0]:
-        dims = [1]
     is_raw, data = get_tf_tensor_data(tensor)
+    # data is [], i.e., empty array
+    if dims == [0] and not is_raw and not data:
+        np_data = np.array([], dtype=map_onnx_to_numpy_type(new_type))
+        return numpy_helper.from_array(np_data, name=name)
+    # data is a scalar
+    if dims == [0] and not is_raw and len(data) == 1:
+        return helper.make_tensor(name, new_type, [], data, False)
+    # which case?
```

Comment:
seems it is for https://github.com/onnx/tensorflow-onnx/pull/162
just remove this comment.